src/sqlfluff/rules/L061.py

```diff
@@ -0,0 +1,54 @@
+"""Implementation of Rule L061."""
+
+from typing import Optional
+
+from sqlfluff.core.parser.segments.raw import CodeSegment
+from sqlfluff.core.rules.base import BaseRule, LintFix, LintResult, RuleContext
+from sqlfluff.core.rules.doc_decorators import document_fix_compatible
+
+
+@document_fix_compatible
+class Rule_L061(BaseRule):
+    """Use ``!=`` instead of ``<>`` for ``not equal to`` comparison.
+
+    | **Anti-pattern**
+    | ``<>`` means ``not equal`` but doesn't sound like this when
+    | we say it loud.
+
+    .. code-block:: sql
+
+        SELECT * FROM X WHERE 1 <> 2;
+
+    | **Best practice**
+    | Use ``!=`` instead because it's sounds more natural and
+    | is more common in other programming languages.
+
+    .. code-block:: sql
+
+        SELECT * FROM X WHERE 1 != 2;
+
+    """
+
+    def _eval(self, context: RuleContext) -> Optional[LintResult]:
+        """Use ``!=`` instead of ``<>`` for ``not equal to`` comparison."""
+        # Only care about not_equal_to segments.
+        if context.segment.name != "not_equal_to":
+            return None
+
+        # Only care about ``<>``
+        if "<>" not in context.segment.raw:
+            return None
```

Comment:
You are correct to scope to `not_equal_to` segment in the first guard clause, but when it comes to detecting the `<>` you'll need some smarter logic to account for the case above.

I would recommend you do something like
```python
raw_comparison_operators = context.functional.segment.children().select(select_if=sp.is_type("raw_comparison_operator"))
if not [r.raw for r in raw_comparison_operators] == ["<", ">"]:
    return None
```
for this second guard clause

Reply:
for god's sake...who does something like this...thanks for your suggestion 🚀  I will look into this tomorrow.

Reply:
N.B. all the grammars parse like this for consistency, so the method mentioned above will work for all dialects 👍

Reply:
Hmmmm.... I'm wondering if this is worth the effort to be honest.

If it's as simple as above, and there are no further issues, then let's go with it. However, if it starts to grow legs than I would favour simplicity over complexity and is some does something like that, then I'm fine with this rule not catching it because of that.

Reply:
I've changed that (based on @jpy-git 's suggestion) because, we need the children of the `not_equal_to` segment for the proper replacement. So it's no big deal.